### PR TITLE
#184637147 retrieve a list of all Products publicly 

### PR DESCRIPTION
**Pull request by katros1**
### What does this PR do?

- [x] All users can be to get all products in stock without signing in.

### How should this be manually tested?

**Installation**
**1. Clone the repository by running** 
```
git clone https://github.com/atlp-rwanda/destructors-ec-bn.git
```
**2. Navigate to the project directory by running cd destructors-ec-bn.**
```
cd destructors-ec-bn
```
then type this in your terminal to navigate to ft-retrieveAllProducts branch
```
git checkout ft-retrieveAllProducts
```
**3. Install the necessary dependencies by running npm install.**
```
npm install
```
**4. run migrations**
```
 npm run migrate
```
**5. run seeds**
```
npm run seed
```
**6. Run the development server by running npm run dev. This will start the server using Nodemon, which will automatically reload the server when changes are made to the code.**
```
npm run dev
```
**7. Run the tests by running npm run test** 
```
npm run test
```
### Any background context you want to provide?

- [x]  open swager in your browser by typing in URL field `localhost:PORT/api-docs`, then under products tag use `api/v1/products/unprotected` to retrieve all products

### Questions:

- none

### CODE REVIEW

File: __test__/product.test.js [hunk 1]
```diff
@@ -119,6 +119,11 @@ describe('Testing Products service', () => {
       .set('Authorization', `Bearer ${token}`);
     expect(response.statusCode).toBe(200);
   });
+  test('it should return 200 for retrieving list of all products for all users', async () => {
+    const response = await request(app)
+      .get('/api/v1/products/unprotected')
```
Comment (Abdulkeza) [inline __test__/product.test.js]: change this route from  '/api/v1/products/unprotected' to '/api/v1/products/public'  at least

File: src/swagger.js [hunk 2]
```diff
@@ -90,6 +91,9 @@ export const swaggerDocument = {
     '/api/v1/products': {
       post: createProduct,
     },
+    '/api/v1/products/unprotected?page={page}&size={size}': {
```
Comment (Abdulkeza) [inline src/swagger.js]: same here, change this route from we can have /api/v1/products/public instead of  api/v1/products/unprotected

**Review by Abdulkeza — CHANGES_REQUESTED** [review]
left feedback bellow

File: __test__/product.test.js (hunk 1 above)
Comment (katros1) [reply __test__/product.test.js]: cool I was think the right word to use , 'public' looks good, thanks

**Pull request merged**